### DISCUSSION TITLE
Constellationship Licensing

### DISCUSSION THREAD

**Discussion by cnlohr** [Stellarium/stellarium#790] [Q&A] [answered]
I'd like to use the constellationship present in Stellarium's constellationship.fab, for a project which will be licensed under the MIT license.  I'd prefer not to have to spend hours recreating the constellationship from another source manually.  I was wondering if the constellationship files for the western constellationships were taken from another source, that I may be able to credit them, or were they specifically written, manually, for Stellarium and licensed under its license and thus unavailable without re-licensing from original author?

**Comment by alex-w** [comment]
No exists an official lines for constellations, but some of lines collections are wide famous - like Ray's or S&T schemes for constellations. The standard Western theme in Stellarium is under GPLv2+, because she is not have separate license for data files. By formal you may generate an own constellationship.fab file from star catalog and some original data - like here: https://github.com/skaven81/western_SnT/ (or with small patches here: https://github.com/Stellarium/stellarium/tree/master/skycultures/western_SnT).

**Comment by cnlohr** [comment]
May I use your constellation lines in my product so long as I give credit to Stellarium in the source?  The stars you have chosen are the most visually appealing.

**Comment by gzotti** [comment]
I think that's OK for us. Funny, other people keep sending us proposals where we made an "error". The stick figures are not standardized, so it's the "Stellarium way" to draw the constellation stick figures.

**Comment by cnlohr** [comment]
It would be interesting to declare a given date from the Stellarium constellationship as a release like "StellarumConstellationship2019."  I think I'd be up for adding it to the Wikipedia entries for the constellations.  I greatly appreciate your consideration for this.  I can imagine being able to have some unrestricted use western constellationship database being particularly in a variety of situations.  Do you have any idea of any other steps that would need to be taken on Stellarium's part to "approve" this for public use?

**Comment by gzotti** [comment]
Your wording is a bit funny IMHO. The name apparently derives from "constellations defined in HIP(parcos) star numbers", and the ending I assume is derived from the project founder's first name. (Team, please correct me if I am off!)  The format looks pretty ad-hoc to me, but it's so simple that it could probably be used by other FOSS projects. (Even the whole "skyculture" concept could be shared, or instead willing  developers should join our project, there is still lots to do! :-)

I have seen Stellarium's constellations used in many circumstances, even the artwork seems to have become a "modern standard". I don't think we have the spare time to re-publish separate snapshots of the same default constellation files elsewhere, on the contrary we are collecting other skycultures. If you like, I am pretty sure you can create screenshots or other representations for WP without licensing conflicts, but please give credit to the Stellarium project, and yes, specify which program version you used, because the line sets are still evolving somewhat. If you use the .fab files directly in your open-source program, I hope GPL2 does not conflict with MIT license. @xalioth , @alex-w , @guillaumechereau , objections?

**Comment by xalioth** [comment]
I'm the author of the constellation lines for the western sky culture. It's licensed under a GPL licence, so in theory you may not use it in a MIT project, but if I give you my permission to re-licence it under MIT (which I happily do!) I guess that's enough? 
This doesn't apply to the art images. For these you need to ask permission to Johan Meuris.

**Comment by cnlohr** [comment]
@gzotti specifically the reason I asked this question is the GPL2 conflicts with the MIT license.  

@xalioth I appreciate this greatly and will include stellarium in the copyright notice.  For my purposes your comment is sufficient.  This chain did turn a little bit into a "what about everyone else?" question :-p.  I.e. what to do about everyone else who was using the line connections.

I will not be maintaining the format of the file as my output will be a stand-alone C file.

Regardless, I will "take this and run" and be sure to:
(1) Include notes regarding this interaction
(2) Include that the source was Stellarium
(3) That the permission is extended only to this data set.
(4) The specific version of Stellarium I pulled it from.

I also intend to update this ticket when I get a video of my project working :)

**Comment by cnlohr** [comment]
Sorry @xalioth I just re-read your comment - you said you were happy to give permission, do you actually grant it :-p

**Comment by xalioth** [comment, marked as answer]
Yes I do grant the permission!!
I hate legal stuff!

**Reply by ArVanDamme** [reply]
Hi, i actually want to create a 3D app using C# to explore universe and i'd like to use it too can i ?

**Reply by gzotti** [reply]
This depends on the kind of program you are developing, and its license conditions. 
* Stellarium is under GPL2.
* As @xalioth has written, the definition of lines (stick figures) for the western skyculture can also be used under MIT license. The artwork requires clarification with Johan Meuris.

**Reply by reubn** [reply]
As per https://johanmeuris.eu/work/stellarium-constellation-art/ Johan Meuris has licensed the constellation artwork under the [Free Art License](http://artlibre.org/licence/lal/en/)

**Comment by cnlohr** [comment]
Thank you!

**Comment by cnlohr** [comment]
Just FYI In case anyone wants to see what I'm messing around with this for... https://twitter.com/cnlohr/status/1182830864209723392
